src/extension.ts

```diff
@@ -387,10 +432,35 @@ export class Extension implements vscode.Disposable {
         vscode.window.onDidChangeTextEditorSelection(e => {
           const mode = this.modeMap.get(e.textEditor.document)
 
-          if (mode === Mode.Insert)
+          if (mode === Mode.Insert) {
             this.setDecorations(e.textEditor, this.insertMode.decorationType)
-          else
+            let reset = true
+            //! If selection kind is Command it should/can be produced by insertion/append from dance
+            //! If selection kind is Keyboard it should/can be produced by textchanges
+            if(!!e.kind && ((e.kind === vscode.TextEditorSelectionChangeKind.Command) || (e.kind === vscode.TextEditorSelectionChangeKind.Keyboard))) {
```

Comment:
```suggestion
            if (e.kind !== undefined && ((e.kind === vscode.TextEditorSelectionChangeKind.Command) || (e.kind === vscode.TextEditorSelectionChangeKind.Keyboard))) {
```

Reply:
I don't like implicit conversions, can you explicitly compare against the value we don't want?